src/OmniSharp.Roslyn.CSharp/Services/Refactoring/RunFixAllCodeActionService.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            _logger = loggerFactory.CreateLogger<RunFixAllCodeActionService>();
+            _fixAllDiagnosticProvider = new FixAllDiagnosticProvider(diagnosticWorker);
+        }
+
+        public async Task<RunFixAllResponse> Handle(RunFixAllRequest request)
+        {
+            if(request.Scope != FixAllScope.Document && request.FixAllFilter == null)
+                throw new NotImplementedException($"Only scope '{nameof(FixAllScope.Document)}' is currently supported when filter '{nameof(request.FixAllFilter)}' is not set.");
+
+            var solutionBeforeChanges = Workspace.CurrentSolution;
+
+            var mappedProvidersWithDiagnostics = await GetDiagnosticsMappedWithFixAllProviders(request.Scope, request.FileName);
+
+            var filteredProvidersWithFix = mappedProvidersWithDiagnostics
+                .Where(diagWithFix =>
+                {
+                    if (request.FixAllFilter == null)
+                        return true;
+
+                    return request.FixAllFilter.Any(x => diagWithFix.HasFixForId(x.Id));
+                });
+
+            foreach (var singleFixableProviderWithDocument in filteredProvidersWithFix)
+            {
+                try
+                {
+                    var document = Workspace.CurrentSolution.GetDocument(singleFixableProviderWithDocument.DocumentId);
+
+                    var fixer = singleFixableProviderWithDocument.FixAllProvider;
+
+                    var (action, fixableDiagnosticIds) = await singleFixableProviderWithDocument.RegisterCodeFixesOrDefault(document);
+
+                    if (action == null)
+                        continue;
+
+                    var fixAllContext = new FixAllContext(
+                        document,
+                        singleFixableProviderWithDocument.CodeFixProvider,
+                        Microsoft.CodeAnalysis.CodeFixes.FixAllScope.Project,
+                        action.EquivalenceKey,
+                        fixableDiagnosticIds,
+                        _fixAllDiagnosticProvider,
+                        CancellationToken.None
+                    );
+
+                    var fixes = await singleFixableProviderWithDocument.FixAllProvider.GetFixAsync(fixAllContext);
+
+                    if (fixes == null)
+                        continue;
+
+                    var operations = await fixes.GetOperationsAsync(CancellationToken.None);
+
+                    foreach (var o in operations)
+                    {
+                        _logger.LogInformation($"Applying operation {o.ToString()} from fix all with fix provider {singleFixableProviderWithDocument.CodeFixProvider} to workspace document {document.FilePath}.");
+
+                        if (o is ApplyChangesOperation applyChangesOperation)
```

Comment:
there is a lot of overlap between what this endpoint does and what the existing run code action does - however the logic there seems a lot more dilligent https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/RunCodeActionService.cs#L100 as it handles added documents too etc. it seems like this code could be reconciled together for better maintenance and correctness?

also, I am not sure the changes should be applied to the workspace here? I think it should be controlled via `ApplyTextChanges` like in the other endpoint

Reply:
@filipw what WantsAllCodeActionOperations stands for?

Reply:
Refactored fixAll use same code to resolve file changes as code action service.

Reply:
Supporting `ApplyTextChanges` seems to be tricky, i think routine that currently generates result must be written ground up to get it work. If i use same kind of solution as in code actions with it batch fixing sometimes works, sometimes not and sometimes it results inconsistent changes. That only occurs when i try "fix all issues in current document", it seems to work ok(ish) when only one type of issue is fixed at time. However fix all issues on save/hotkey is very nice feature to support initially 🤔

I believe reason behind for those problems is that after every type new dianostics are used, however diagnostics are based on workspace state, not solution currently half-updated between routines which messes up things (diagnostics can point incorrect locations at file etc). That is pretty tricky thing to implement correctly. Is it ok if `ApplyTextChanges` support is postponed until someone truly likes to use it? I think i will return with to this thing later on once i have heureka moment how to actually do it correctly. That solution probably also gives possibility to support "fix everything at project" and "fix everything at solution" that had similar kind of issues and didn't suppor them for that reason at first version.